spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java

```diff
@@ -0,0 +1,222 @@
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestReplaceBranch extends SparkExtensionsTestBase {
+  @Parameterized.Parameters(name = "catalogName = {0}, implementation = {1}, config = {2}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+      {
+        SparkCatalogConfig.SPARK.catalogName(),
+        SparkCatalogConfig.SPARK.implementation(),
+        SparkCatalogConfig.SPARK.properties()
+      }
+    };
+  }
+
+  public TestReplaceBranch(String catalogName, String implementation, Map<String, String> config) {
+    super(catalogName, implementation, config);
+  }
+
+  @After
+  public void removeTable() {
+    sql("DROP TABLE IF EXISTS %s", tableName);
+  }
+
+  @Test
+  public void testReplaceBranchFailsForTag() throws NoSuchTableException {
+    sql("CREATE TABLE %s (id INT, data STRING) USING iceberg", tableName);
+    String tagName = "tag1";
+
+    List<SimpleRecord> records =
+        ImmutableList.of(new SimpleRecord(1, "a"), new SimpleRecord(2, "b"));
+    Dataset<Row> df = spark.createDataFrame(records, SimpleRecord.class);
+    df.writeTo(tableName).append();
+    Table table = validationCatalog.loadTable(tableIdent);
+    long first = table.currentSnapshot().snapshotId();
+    table.manageSnapshots().createTag(tagName, first).commit();
+    df.writeTo(tableName).append();
+    long second = table.currentSnapshot().snapshotId();
+
+    AssertHelpers.assertThrows(
+        "Cannot perform replace branch on tags",
+        IllegalArgumentException.class,
+        "Ref tag1 is a tag not a branch",
+        () -> sql("ALTER TABLE %s REPLACE BRANCH %s AS OF VERSION %d", tableName, tagName, second));
+  }
+
+  @Test
+  public void testReplaceBranch() throws NoSuchTableException {
+    sql("CREATE TABLE %s (id INT, data STRING) USING iceberg", tableName);
+    List<SimpleRecord> records =
+        ImmutableList.of(new SimpleRecord(1, "a"), new SimpleRecord(2, "b"));
+    Dataset<Row> df = spark.createDataFrame(records, SimpleRecord.class);
+    df.writeTo(tableName).append();
+
+    Table table = validationCatalog.loadTable(tableIdent);
+    long first = table.currentSnapshot().snapshotId();
+    String branchName = "b1";
+    table.manageSnapshots().createBranch(branchName, first).commit();
+    SnapshotRef b1 = table.refs().get(branchName);
+    Integer minSnapshotsToKeep = b1.minSnapshotsToKeep();
+    Long maxSnapshotAgeMs = b1.maxSnapshotAgeMs();
+    Long maxRefAgeMs = b1.maxRefAgeMs();
+    df.writeTo(tableName).append();
+    long second = table.currentSnapshot().snapshotId();
+
+    sql("ALTER TABLE %s REPLACE BRANCH %s AS OF VERSION %d", tableName, branchName, second);
+
+    table.refresh();
+    SnapshotRef ref = table.refs().get(branchName);
+    Assert.assertNotNull(ref);
+    Assert.assertEquals(ref.snapshotId(), second);
+    Assert.assertEquals(minSnapshotsToKeep, ref.minSnapshotsToKeep());
```

Comment:
I wasn't sure the behavior of this - if we are replacing a branch that had set options like min snapshots to keep etc, do we continue to inherit those options or do we use default? in this test we seem to not distinguish these cases

Reply:
When replacing a branch we preserve the existing retention properties. https://github.com/apache/iceberg/blob/c248fe29bc9d52741e69ee425dfccdf7edb99f83/core/src/main/java/org/apache/iceberg/UpdateSnapshotReferencesOperation.java#L104

But good point on the test cases, right now when setting up the original branch, I don't set any retention properties so it's just the default. It's worth updating the tests to have some non-default retention properties to begin with and make sure those are preserved

Reply:
sorry I saw your response that this and my other comment got addressed, but I don't think I see them from this PR, did I miss anything?

Reply:
Sorry for the confusion, I somehow missed updating the test. I updated testReplaceBranch, the initial branch which gets created has non-default retention properties, and then after the replace is performed an assertion on the expected retention properties (the original retention properties at creation time) is done on the updated reference.